README.rst

```diff
@@ -209,6 +209,39 @@ The underlying reason is, that services like *Read the Docs* sometimes change
 the working directory for good reasons and using the installed metadata
 prevents using needless volatile data there.
 
+Usage from Docker
+-----------------
+
+By default, docker will not copy the ``.git``  folder into your container.
+Therefore, builds with version inference might fail.
+Consequently, you can use the following snipped to infer the version from
+the host os without copying the entire ``.git`` folder to your Dockerfile.
+
+.. code:: dockerfile
+
+    RUN --mount=source=.git,target=.git,type=bind \
+        pip install --no-cache-dir -e .
+
+However, this build step introduces a dependency to the state of your local
+.git folder the build cache and triggers the long-running pip install process on every build.
+To optimize build caching, one can use an environment variable to pretend a pseudo
+version that is used to cache the results of the pip install process:
+
+.. code:: dockerfile
+
+    FROM python
+    COPY pyproject.toml
+    ARG PSEUDO_VERSION=1
+    RUN SETUPTOOLS_SCM_PRETEND_VERSION=${PSEUDO_VERSION} pip install -e .[test]
```

Comment:
random thought, would it help to bind mount the pip cache of the invoking user?

Reply:
Yes, some people do that. https://stackoverflow.com/a/58021389
However, is this specific to setuptools? Maybe one can link general recommendations for working with docker and python. I think both options, mounting .git and using the pretend version environment variables are specific for setuptools. The rest depends on the specific use case with pros and cons.
The line in https://github.com/pypa/setuptools_scm/issues/568#issue-896566554 is to my feeling the important bit to share, and effective if one does not care about waiting for pip install to complete. If you like, feel free to remove the second part with the pretend version altogether. I am uncertain, if this is still in the scope of #568.

Reply:
thanks, im outside for a while longer then i'l ltake a look at mergin this in the evening